java/src/hu/holyoil/view/panels/InventoryListPanel.java

```diff
@@ -1,27 +1,202 @@
+
+        inventory.setLayoutOrientation(JList.VERTICAL);
+        inventory.setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
+        inventory.setBackground(new Color(4,4,13)); //lista háttere (nem lehet átlátszó)
+        inventory.setCellRenderer(new InventoryCellRenderer());
+
+        JScrollPane scrollPane = new JScrollPane(inventory);
+        scrollPane.setVerticalScrollBar(new JScrollBar());
+        scrollPane.setBackground(new Color(4,4,13));
+        scrollPane.setPreferredSize(new Dimension(320, 280));
 
+        JLabel tpText = new JLabel("Number of TeleportGates: ");
+        tps= new JLabel(String.valueOf(storage.GetTeleporterCount()));
+        craftRobot = new JButton("Craft Robot");
+        craftTp = new JButton("Craft Teleporter");
+        placeTp = new JButton("Place Teleporter");
+        fill = new JButton("Place Resource");
+        //első két gomb
+        JPanel panel1 = new JPanel();
+        panel1.setOpaque(false);
+        panel1.add(fill);
+        fill.setEnabled(false);
+        panel1.add(placeTp);
+        //teleporter szöveg és alsó két gomb
+        JPanel panel2 = new JPanel();
+        panel2.setOpaque(false);
+        panel2.setLayout(new BoxLayout(panel2, BoxLayout.Y_AXIS));
+        JPanel panel3 = new JPanel();
+        panel3.setOpaque(false);
+        panel3.add(tpText);
+        panel3.add(tps);
+        JPanel panel4 = new JPanel();
+        panel4.setOpaque(false);
+        panel4.add(craftRobot);
+        panel4.add(craftTp);
+        panel2.add(panel3);
+        panel2.add(panel4);
+
+        GridBagConstraints c = new GridBagConstraints();
+        c.fill = GridBagConstraints.HORIZONTAL;
+        c.gridy=0; c.gridx=0;
+        c.gridwidth=360;
+
+        add(scrollPane, c);
+        c.gridy=1;
+        add(panel1, c);
+        c.gridy=2;
+        add(panel2, c);
+
+        setOpaque(false);
+        setVisible(true);
     }
 
+    /**
+     * Listenereket ad a gombokhoz
+     */
     private void InitListeners() {
-
+        craftRobot.addActionListener(e -> settler.CraftRobot());
+        craftTp.addActionListener(e -> settler.CraftTeleportGate());
+        placeTp.addActionListener(e -> settler.PlaceTeleporter());
+        fill.addActionListener(e -> {
+            if(!inventory.isSelectionEmpty()) settler.PlaceResource(inventory.getSelectedValue());
+        });
+        inventory.addListSelectionListener(e -> fill.setEnabled(!(inventory.isSelectionEmpty())
+                && settler.GetOnAsteroid().GetResource()==null
+                && settler.GetOnAsteroid().GetLayerCount()==0)
+        );
     }
 
+    /**
+     * Újratölti a körök végén a tárhelyeket.
+     */
     @Override
     public void UpdateComponent() {
-
+        Asteroid current= settler.GetOnAsteroid();
+        InitListeners();
+        placeTp.setEnabled(storage.GetTeleporterCount()!=0 && current.GetTeleporter()==null);
+        settler = TurnController.GetInstance().GetSteppingSettler();
```

Comment:
This should be at the beginnig of the function. Currently it checks the inventory and asteroid of the last settler